### ISSUE TITLE
[B1.2] Implement RMSprop optimizer in shared/autograd/optimizers.mojo

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#2299]
## Overview
Implement the RMSprop optimizer for adaptive learning rates.

## Location
- `shared/autograd/optimizers.mojo` (TODO already exists at line 156)

## Algorithm
RMSprop divides learning rate by running average of gradient magnitudes:
```
v_t = ρ * v_{t-1} + (1 - ρ) * g_t²    # Running average
θ_t = θ_{t-1} - α * g_t / (√v_t + ε)  # Update
```

## Deliverables
- [ ] Implement RMSprop struct with:
  - learning_rate (default: 0.01)
  - alpha/rho (default: 0.99) - smoothing constant
  - epsilon (default: 1e-8)
  - weight_decay (default: 0.0)
  - momentum (default: 0.0) - optional momentum
- [ ] Implement step() method matching SGD interface
- [ ] Implement zero_grad() method
- [ ] Store squared gradient buffers per parameter
- [ ] Export from `shared/autograd/__init__.mojo`

## Success Criteria
- RMSprop optimizer works with Variable/GradientTape system
- Training converges on test problems
- CI passes

## Dependencies
- None (parallel with B1.1, B1.3)

## Track
Track B: Missing Features (B1.2 of 8)
